tests/integration/test_data_store_mgr.py

```diff
@@ -313,12 +344,93 @@ async def test_update_data_structure(harness):
     assert TASK_STATUS_FAILED in set(collect_states(data, FAMILY_PROXIES))
     # state totals changed
     assert TASK_STATUS_FAILED in data[WORKFLOW].state_totals
-    # Shows pruning worked
-    # TODO: fixme
-    # https://github.com/cylc/cylc-flow/issues/4175#issuecomment-1025666413
-    # assert len({t.is_held for t in data[TASK_PROXIES].values()}) == 1
 
 
+async def test_prune_data_store(harness_prune):
+    """Test prune_data_store. This method will expand and reduce the data-store
+    to invoke pruning."""
+    schd, data = harness_prune
+    w_id = schd.data_store_mgr.workflow_id
+    schd.data_store_mgr.data[w_id] = data
+    schd.pool.hold_tasks(['*'])
```

Comment:
This test is mutating the task pool so it should not be using the module scoped `harness_prune` fixture (or make `harness_prune` function scoped)

Reply:
I've removed this extra module scoped `harness_prune`.. 
Changed the pool modifying prune test to have it's own scheduler to toy with, and shifted the non-pool modifying test to the existing `harness`.